src/connection.rs

```diff
@@ -75,10 +75,10 @@ impl Connection {
 			}
 		}
 
-		let (tx, rx) = mpsc::channel();
+		let (tx, mut rx) = mpsc::channel();
 		try!(::std::thread::Builder::new()
 			.name("Discord Keepalive".into())
-			.spawn(move || keepalive(heartbeat_interval, sender, rx)));
+			.spawn(move || keepalive(heartbeat_interval, sender, &mut rx)));
```

Comment:
I want them to be, exclude this (and the other `mpsc::Receiver` changes) and everything else should be good.